weblate/trans/views/edit.py

```diff
@@ -638,27 +638,26 @@ def translate(request: AuthenticatedHttpRequest, path):
             messages.error(request, gettext("Invalid search string!"))
             return redirect(obj)
 
-    from_offset = int(request.GET.get("from_offset", offset - 1))
-    if from_offset and 1 <= from_offset <= len(search_result["ids"]):
-        previous_unit = unit_set.get(pk=search_result["ids"][from_offset - 1])
-
+    previous_unit_id = int(request.session['previous_unit_id']) if 'previous_unit_id' in request.session else None
```

Comment:
This should be pinned to the search result, otherwise it will have too many corner cases:

* Translating one project, then starting translating another project would yield the message.
* Opening multiple translation tabs they will fight for the same previous_unit_id.

It could stored inside the search data which we already have in the `search` function.